lib/api/completeMultipartUpload.js

```diff
@@ -221,17 +213,38 @@ function completeMultipartUpload(authInfo, request, log, callback) {
                 }
                 // Assemble array of part locations, aggregate size and build
                 // string to create aggregate ETag
-                const partRetrievalInfo = {
-                    key: storedPartsAsObjects[i].location,
-                    size: partSize,
-                    start: calculatedSize,
-                    dataStoreName: storedPartsAsObjects[i].dataStoreName,
-                    cryptoScheme: storedPartsAsObjects[i].sseCryptoScheme,
-                    cipheredDataKey: storedPartsAsObjects[i].sseCipheredDataKey,
-                };
 
-                dataLocations.push(partRetrievalInfo);
-                calculatedSize += partSize;
+                // If part was put by a regular put part rather than a copy
+                // it is always one location.  With a put part copy, could
+                // be multiple locations so loop over array of locations.
+                for (let j = 0; j < storedPartsAsObjects[i].locations.length;
```

Comment:
I spoke with @vrancurel about this.  It's unlikely someone has an open MPU from that far back so it would be a good time to make this change rather than complicating the code.